### PR TITLE
Fix tsan issues

### PR DESCRIPTION
**Pull request by hassanctech**
This doesn't address all the issues reported by TSAN but the ones addressed so far are:

(1) In getStreamingEndpointCachingCurl we were not checking if a mutex was locked before unlocking, check added.
(2) In T1 we call `createDeviceCurl` which grabs the `activeRequestsLock` then we call `createDeviceResultEvent` in PIC which calls `createDeviceResult` which acquired the client lock.  In T2 in `putFrame` we first acquire the stream lock, then the client lock.  Next we invoke the callback in producer-c `streamLatencyPressureFn` which eventually calls `getStreamingEndpointCurl` which now acquires the `activeRequestsLock`.  This could result in deadlock and was the *only* potential deadlock reported by TSAN.  I think we never hit this because no one uses create Device.  Nonetheless to remedy this I've removed the acquisition of `activeRequestsLock` by T1 in `createDeviceCurl`

### CODE REVIEW

**Review by unicornss — APPROVED** [review]
looks good.

File: CMake/Dependencies/libkvspic-CMakeLists.txt [hunk 1]
```diff
@@ -7,7 +7,7 @@ include(ExternalProject)
 # clone repo only
 ExternalProject_Add(libkvspic-download
 	GIT_REPOSITORY    https://github.com/awslabs/amazon-kinesis-video-streams-pic.git
-    	GIT_TAG           76223690da1329d733309e2a52575d6a98a68981
```
Comment (disa6302) [inline CMake/Dependencies/libkvspic-CMakeLists.txt:10]: Unintended change since there is a PR for it already?

**Review by disa6302 — APPROVED** [review]
Rest LGTM.

File: CMake/Dependencies/libkvspic-CMakeLists.txt (hunk 1 above)
Comment (hassanctech) [reply CMake/Dependencies/libkvspic-CMakeLists.txt:10]: Yeah at the time I put out this PR, the other one hadn't been merged -- I'll update branch before merging.

**Pull request merged**